### PR TITLE
Story/hfi ready toggle db/1859

### PR DESCRIPTION
**Pull request by conbrad**
Includes all feedback from previous PR: https://github.com/bcgov/wps/pull/2049

#### New table schema:
```
hfi_ready
- id (uuid, PK)
- hfi_request_id (FK, hfi_request)
- planning_area_id (FK, planning_area)
- ready (boolean)
- create_timestamp (TZ)
- create_user (idir string)
- update_timestamp (TZ)
- update_user (idir string)
```
PK is a UUID because a lot of records are gonna be generated on HFI request changes.
#### Behaviour changes:

Updating a `hfi_request` (fire starts, fuel type, etc), creates a new `hfi_request` and needs to copy over the `hfi_ready` record data, from the updated `hfi_request`, into new `hfi_ready` records that point to the new `hfi_request`

**_Next PR will hook the changes up to responses and handle ready state toggles through API._**

# Test Links:
[Percentile Calculator](https://wps-pr-2072.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-2072.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-2072.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-2072.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-2072.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Fire Behaviour Advisory](https://wps-pr-2072.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-2072.apps.silver.devops.gov.bc.ca/hfi-calculator)
[FWI Calculator](https://wps-pr-2072.apps.silver.devops.gov.bc.ca/fwi-calculator)

### CODE REVIEW

File: api/app/db/crud/hfi_calc.py [hunk 1]
```diff
@@ -77,14 +77,62 @@ def get_most_recent_updated_hfi_request_for_current_date(session: Session,
...
+    now = get_utc_now()
+
+    if previous_latest_request is not None:
+        # Copy over ready records for existing hfi request
+        latest_hfi_ready_records = get_latest_hfi_ready_records(session, previous_latest_request.id)
+        updated_hfi_ready_records = []
+        for latest_hfi_ready_record in latest_hfi_ready_records:
+            updated_hfi_ready_records.append(HFIReady(hfi_request_id=latest_hfi_request_id,
+                                                      planning_area_id=latest_hfi_ready_record.planning_area_id,
+                                                      ready=latest_hfi_ready_record.ready,
```
Comment (Sybrand) [inline api/app/db/crud/hfi_calc.py:103]: If the request is being updated - should the ready state not be set to false depending on which planning area is being updated?

**Review by Sybrand — APPROVED** [review]
Just a question around ready being set false on updates. Otherwise looks good. (biting my tongue on uuid....)

File: api/app/db/crud/hfi_calc.py (hunk 1 above)
Comment (conbrad) [reply api/app/db/crud/hfi_calc.py:103]: I'm not sure to be honest so I chose to keep the behaviour that adds less complexity.

Comment (Sybrand) [reply api/app/db/crud/hfi_calc.py:103]: No worries - I'm sure it will surface as a bug once we start toggling values on/off.

Comment (conbrad) [reply api/app/db/crud/hfi_calc.py:103]: Actually, good call, it was in additional context: "logic: if a user changes fire starts > marks as ready > then excludes a station, changes a fire start, or a fuel type > we switch toggle back to 'not ready'"

I'll make the change

**Review by andrea-williams — APPROVED** [review]

File: api/app/db/crud/hfi_calc.py (hunk 1 above)
Comment (conbrad) [reply api/app/db/crud/hfi_calc.py]: Done

**Pull request merged**